v3/src/components/graph/graph-component-handler.ts

```diff
@@ -84,7 +84,12 @@ export const graphComponentHandler: DIComponentHandler = {
     const layers: Array<IGraphPointLayerModelSnapshot> = []
     let provisionalDataSet: IDataSet | undefined
     let provisionalMetadata: IDataSetMetadata | undefined
-    getSharedDataSets(appState.document).forEach(sharedDataSet => {
+    const sharedDataSets = getSharedDataSets(appState.document)
+    // We currently only support one dataset in a graph, so we find the one specified by plugin
```

Comment:
Also, if no `DataSet` is specified in the API and there's only one `DataSet` in the document, we should default to it (as graphs created within CODAP by the user do).